### PR TITLE
[Yahoo/NBCSports] Fix 5226 and add support for NBC sports

### LINKED ISSUE #5226: ERROR: Yahoo returned error: missing video tag
URL: http://sports.yahoo.com/blogs/ncaab-the-dagger/tyler-kalinoski-s-buzzer-beater-caps-davidson-s-comeback-win-185609842.html?guid=nbc_cbk_davidsonbuzzerbeater_150313

C:\>youtube-dl -v "http://sports.yahoo.com/blogs/ncaab-the-dagger/tyler-kalinosk
i-s-buzzer-beater-caps-davidson-s-comeback-win-185609842.html?guid=nbc_cbk_david
sonbuzzerbeater_150313"
[debug] System config: []
[debug] User config: []
[debug] Command-line args: ['-v', 'http://sports.yahoo.com/blogs/ncaab-the-dagge
r/tyler-kalinoski-s-buzzer-beater-caps-davidson-s-comeback-win-185609842.html?gu
id=nbc_cbk_davidsonbuzzerbeater_150313']
[debug] Encodings: locale cp1252, fs mbcs, out cp850, pref cp1252
[debug] youtube-dl version 2015.03.15
[debug] Python version 2.7.8 - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg N-40824-g31dfe20
[debug] Proxy map: {}
[Yahoo] tyler-kalinoski-s-buzzer-beater-caps-davidson-s-comeback-win: Downloadin
g webpage
[Yahoo] tyler-kalinoski-s-buzzer-beater-caps-davidson-s-comeback-win: Downloadin
g video info
ERROR: Yahoo returned error: missing video tag
Traceback (most recent call last):
  File "youtube_dl\YoutubeDL.pyo", line 654, in extract_info
  File "youtube_dl\extractor\common.pyo", line 275, in extract
  File "youtube_dl\extractor\yahoo.pyo", line 172, in _real_extract
  File "youtube_dl\extractor\yahoo.pyo", line 195, in _get_info
ExtractorError: Yahoo returned error: missing video tag

Thanks Ringo

### PR DESCRIPTION
**Pull request by yan12125**
This should fix #5226 

I also addd an ```NBCSportsIE```. ```YahooIE``` and ```NBCSportsIE``` are both based on ```NBCSportsVPlayerIE```. I put ```NBCSportsIE``` and ```NBCSportsVPlayerIE``` before ```NBCNewsIE``` because they're more relevant to ```NBCIE```.

### CODE REVIEW

File: youtube_dl/extractor/yahoo.py [hunk 1]
```diff
@@ -151,6 +163,10 @@ def _real_extract(self, url):
                 items = json.loads(items_json)
                 video_id = items[0]['id']
                 return self._get_info(video_id, display_id, webpage)
+        # Look for NBCSports iframes
+        nbc_sports_url = NBCSportsVPlayerIE._extract_url(webpage)
+        if nbc_sports_url:
+            return self.url_result(nbc_sports_url, 'NBCSportsVPlayer')
```
Comment (dstftw) [inline youtube_dl/extractor/yahoo.py]: This should be placed in generic extractor as well. vplayer is a generic embed (e.g. http://bbs.clutchfans.net/showthread.php?t=244180)

File: youtube_dl/extractor/nbc.py [hunk 2]
```diff
@@ -50,6 +50,56 @@ def _real_extract(self, url):
         return self.url_result(theplatform_url)
 
 
+class NBCSportsVPlayerIE(InfoExtractor):
+    _VALID_URL = r'https?://vplayer\.nbcsports\.com/(?:[^/]+/)+(?P<id>[0-9a-zA-Z]+)'
```
Comment (dstftw) [inline youtube_dl/extractor/nbc.py]: Looks like `_` should be allowed for `id` (e.g. http://vplayer.nbcsports.com/p/BxmELC/nbc_embedshare/select/_hqLjQ95yx8Z)

**Comment by yan12125** [comment]
@dstftw Thanks for the comments. The video in http://bbs.clutchfans.net/showthread.php?t=244180 is expired (When I play, a woman voice says "Its content is currently unavailable") and I managed to detect such errors. Is there a page containing valid NBCSports videos?

**Comment by yan12125** [comment]
Seems ```Test_NBCSports``` and relevant tests gives different MD5 sums each time. I got ```ba6c93f96b67bf05344f78bd523dac0f``` yesterday and Travis CI got ```cf00ead632923826ced9687763fb3ffe```, while I just got ```455cf50acb1339930ce68af8cc8448cc``` on my device several minutes ago. I just comment out all md5 checks before the problem gets solved.

**Comment by dstftw** [comment]
Thanks.

**Pull request merged**